metrics/proxy.go

```diff
@@ -0,0 +1,46 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type ProxyMetrics struct {
+	collectors []prometheus.Collector
+
+	proxyLabels        []string
+	ReceivedBytesTotal *prometheus.CounterVec
+	SentBytesTotal     *prometheus.CounterVec
+}
+
+func (c *ProxyMetrics) Collectors() []prometheus.Collector {
+	return c.collectors
+}
+
+func NewProxyMetrics() *ProxyMetrics {
```

Comment:
My Go Spidey sense tells me to basically never use pointers for structs unless the struct itself is pretty huge.

This struct is just 4 pointers on its own, so not "pretty huge". Also doesn't seem like it's heavily mutated itself, so probably doesn't need to have interfaces on the pointer itself. 

Maybe.

Reply:
The method `NewStruct` usually returns pointer, otherwise it is a public struct that could be used without initialization method.